### PR TITLE
Feature/lww 258 withdraw crypto

### PR DESCRIPTION
**Pull request by andreyzhylin**

### CODE REVIEW

File: src/pages/ConfirmOperationPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  private handleTryAgain = async (e: any) => {
+    e.preventDefault();
+    const operationId = await this.withdrawStore.sendWithdrawCryptoRequest(
+      this.withdrawStore.withdrawCrypto.assetId,
+      this.withdrawStore.withdrawCrypto
+    );
+    this.props.history.replace(ROUTE_CONFIRM_OPERATION_ID(operationId));
+  };
+
+  private handleSubmit = async () => {
```
Comment (PaulRumyantsevItr) [inline src/pages/ConfirmOperationPage/index.tsx]: Function do too much. Please split it into number of smaller functions

File: src/pages/ConfirmOperationPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      [OpStatus.Confirmed]: () => {
+        this.props.history.replace(ROUTE_WITHDRAW_CRYPTO_SUCCESS);
+      },
+      [OpStatus.Failed]: () => {
+        this.props.history.replace(ROUTE_WITHDRAW_CRYPTO_FAIL);
+      }
+    };
+
+    const TIMEOUT_LIMIT = 10000;
+    const timeout = window.setTimeout(async () => {
```
Comment (PaulRumyantsevItr) [inline src/pages/ConfirmOperationPage/index.tsx]: Name is too common, I think it make sense to rename this handle variable into something more specific

File: src/pages/WithdrawCryptoPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+  readonly withdrawStore = this.props.rootStore!.withdrawStore;
+  readonly walletStore = this.props.rootStore!.walletStore;
+  readonly profileStore = this.props.rootStore!.profileStore;
+  readonly socketStore = this.props.rootStore!.socketStore;
+
+  @computed
+  get balance() {
+    const {assetId} = this.props.match.params;
+
+    if (this.walletStore.tradingWallets.length) {
```
Comment (PaulRumyantsevItr) [inline src/pages/WithdrawCryptoPage/index.tsx]: Convert into early return is wallets are empty

File: src/pages/WithdrawCryptoPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+      const balance = this.walletStore.tradingWallets[0].balances.find(
```
Comment (PaulRumyantsevItr) [inline src/pages/WithdrawCryptoPage/index.tsx]: balance -> balanceModel

File: src/pages/WithdrawCryptoPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+        return balance.balance;
+      }
+    }
+
+    return 0;
+  }
+
+  componentDidMount() {
+    const {assetId} = this.props.match.params;
+    this.withdrawStore.fetchFee(assetId);
```
Comment (PaulRumyantsevItr) [inline src/pages/WithdrawCryptoPage/index.tsx]: Should it be await this.widthdrawStore... ?

File: src/pages/WithdrawCryptoPage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+              onSubmit={this.handleSubmit}
+              render={this.renderForm}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private handleSubmit = async (
```
Comment (PaulRumyantsevItr) [inline src/pages/WithdrawCryptoPage/index.tsx]: Split into smaller parts

File: src/pages/WithdrawCryptoPage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+      setSubmitting(false);
+    }
+  };
+
+  private getFeeSize(amount: number) {
+    const {assetId} = this.props.match.params;
+    const asset = this.assetStore.getById(assetId);
+    return moneyRound(
+      this.withdrawStore.absoluteFee ||
+        this.withdrawStore.relativeFee ||
```
Comment (PaulRumyantsevItr) [inline src/pages/WithdrawCryptoPage/index.tsx]: Looks like a problem with formatting

**Comment by PaulRumyantsevItr** [comment]
Does not see any new tests

File: src/pages/WithdrawCryptoPage/index.tsx (hunk 5 above)
Comment (andreyzhylin) [reply src/pages/WithdrawCryptoPage/index.tsx]: We can't await inside `componentDidMount` :(

**Pull request merged**